### ISSUE TITLE
dev-exp: LZ section 4: project id/number is ambiguous - do we mean project name/id - should be the same for unique cc project previously generated

### ISSUE THREAD

**Issue by obriensystems** [GoogleCloudPlatform/pubsec-declarative-toolkit#98]
section 4 of https://github.com/GoogleCloudPlatform/pubsec-declarative-toolkit/tree/main/solutions/landing-zone#usage
dev-experience: LZ section 4: project id/number is ambiguous - do we mean project name/id - should be the same for unique cc project previously generated

4) we need an example run in the docs
what is project number - I would expect it to be project id but we already have one of these - are they switched as in project name and project id (not always the same unless name is unique - or is it the random postfix
original
```
  management-project-id: management-project-12345
  management-project-number: "0000000000"
```
Why is one in quotes (i need to check their usage)
I would expect the same as the project id should be unique 
modified
```
  management-project-id: landing-zone-controller-1z583
  management-project-number: "landing-zone-controller-1z583"
```

resulting run has issues connecting to the gke cluster - different issue
```
5.deploy

michael@cloudshell:~/github/GoogleCloudPlatform/pubsec-declarative-toolkit/landing-zone (landing-zone-controller-1z583)$ kpt fn render
Package "landing-zone/environments/common/guardrails-policies":
Package "landing-zone/environments/common":
[RUNNING] "gcr.io/kpt-fn/set-namespace:v0.4.1"
[PASS] "gcr.io/kpt-fn/set-namespace:v0.4.1" in 2.1s
  Results:
    [info]: namespace "common" updated to "config-control", 23 value(s) changed

Package "landing-zone/environments/nonprod":
[RUNNING] "gcr.io/kpt-fn/set-namespace:v0.4.1"
[PASS] "gcr.io/kpt-fn/set-namespace:v0.4.1" in 300ms
  Results:
    [info]: namespace "nonprod" updated to "config-control", 7 value(s) changed

Package "landing-zone/environments/prod":
[RUNNING] "gcr.io/kpt-fn/enable-gcp-services:v0.1.0"
[FAIL] "gcr.io/kpt-fn/enable-gcp-services:v0.1.0" in 700ms
  Stderr:
    "docker: Error response from daemon: Get \"https://gcr.io/v2/\": dial tcp [2607:f8b0:400c:c13::52]:443: connect: cannot assign requested address."
    "See 'docker run --help'."
  Exit code: 125

checking auth
```

**Comment by cartyc** [comment]
Will have to update the documents to better differentiate Project ID's and Project Numbers.  
Project id's are a  globally unique identifier for your project and in this case would be "landing-zone-controller-1z583" while a Project number is an automatically generated unique identifier for your project which would be a number. 

To get the project number you grab it from the project dashboard. https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects

**Comment by fmichaelobrien** [comment]
Very good, I have not required the number yet - thought we were using the ID as the PK, GCP uses a composite key under the covers
I see now project number is essentially the primary key LONG, so we have the triplet number, id, name

Rerunning #99

**Comment by fmichaelobrien** [comment]
will adjust the docs
https://console.cloud.google.com/home/dashboard?project=landing-zone-controller-1z583
<img width="736" alt="Screen Shot 2022-09-09 at 8 01 22 AM" src="https://user-images.githubusercontent.com/94715080/189345474-de915b7b-4952-4100-86bf-30baafeaf47d.png">

```
michael@cloudshell:~ (landing-zone-controller-1z583)$ gcloud projects describe landing-zone-controller-1z583
createTime: '2022-09-09T00:24:01.131Z'
labels:
  created-with-arete: 'true'
lifecycleState: ACTIVE
name: landing-zone-controller-1z583
parent:
  id: '41160724790'
  type: folder
projectId: landing-zone-controller-1z583
projectNumber: '453474601356'
```

**Comment by fmichaelobrien** [comment]
results
```
```
michael@cloudshell:~/github/GoogleCloudPlatform/pubsec-declarative-toolkit/landing-zone (landing-zone-controller-1z583)$ kpt fn render
Package "landing-zone/environments/common/guardrails-policies":
Package "landing-zone/environments/common":
[RUNNING] "gcr.io/kpt-fn/set-namespace:v0.4.1"
[PASS] "gcr.io/kpt-fn/set-namespace:v0.4.1" in 2.1s
  Results:
    [info]: namespace "common" updated to "config-control", 23 value(s) changed

Package "landing-zone/environments/nonprod":
[RUNNING] "gcr.io/kpt-fn/set-namespace:v0.4.1"
[PASS] "gcr.io/kpt-fn/set-namespace:v0.4.1" in 300ms
  Results:
    [info]: namespace "nonprod" updated to "config-control", 7 value(s) changed

Package "landing-zone/environments/prod":
[RUNNING] "gcr.io/kpt-fn/enable-gcp-services:v0.1.0"
[PASS] "gcr.io/kpt-fn/enable-gcp-services:v0.1.0" in 2.9s
  Results:
    [info] serviceusage.cnrm.cloud.google.com/v1beta1/Service/config-control/prod-nethost-service-compute: generated service
    [info] serviceusage.cnrm.cloud.google.com/v1beta1/Service/config-control/prod-nethost-service-logging: generated service
[RUNNING] "gcr.io/kpt-fn/set-namespace:v0.4.1"
[PASS] "gcr.io/kpt-fn/set-namespace:v0.4.1" in 400ms
  Results:
    [info]: namespace "prod" updated to "config-control", 4 value(s) changed

Package "landing-zone":
[RUNNING] "gcr.io/kpt-fn/apply-setters:v0.2"
[PASS] "gcr.io/kpt-fn/apply-setters:v0.2" in 1.9s
  Results:
    [info] metadata.annotations.cnrm.cloud.google.com/organization-id: set field value to "583675367868"
    [info] metadata.annotations.cnrm.cloud.google.com/organization-id: set field value to "583675367868"
    [info] spec.projectID: set field value to "net-perimeter-prj-common-old1"
    [info] spec.parentRef.external: set field value to "583675367868"
    ...(87 line(s) truncated, use '--truncate-output=false' to disable)
[RUNNING] "gcr.io/kpt-fn/generate-folders:v0.1.1"
[PASS] "gcr.io/kpt-fn/generate-folders:v0.1.1" in 4.5s
[RUNNING] "gcr.io/kpt-fn/enable-gcp-services:v0.1.0"
[PASS] "gcr.io/kpt-fn/enable-gcp-services:v0.1.0" in 1.7s
  Results:
    [info] serviceusage.cnrm.cloud.google.com/v1beta1/Service/config-control/nonprod-nethost-service-compute: generated service
    [info] serviceusage.cnrm.cloud.google.com/v1beta1/Service/config-control/nonprod-nethost-service-dns: generated service
    [info] serviceusage.cnrm.cloud.google.com/v1beta1/Service/config-control/nonprod-nethost-service-logging: generated service
    [info] serviceusage.cnrm.cloud.google.com/v1beta1/Service/config-control/prod-nethost-service-compute: recreated service
    ...(3 line(s) truncated, use '--truncate-output=false' to disable)
[RUNNING] "gcr.io/kpt-fn/gatekeeper:v0.2.1"
[PASS] "gcr.io/kpt-fn/gatekeeper:v0.2.1" in 3.1s
[RUNNING] "gcr.io/kpt-fn/kubeval:v0.3.0"
[PASS] "gcr.io/kpt-fn/kubeval:v0.3.0" in 21.3s

Successfully executed 9 function(s) in 5 package(s).
```
```

**Comment by cartyc** [comment]
Ya you don't run into project numbers too much. In our case it's part of the Service Account name that's used by Config Controller and some other default system Accounts. For example `serviceAccount:service-${management-project-number}@gcp-sa-yakima.iam.gserviceaccount.com` where the mangement project is where your config controller instance resides.

**Comment by cartyc** [comment]
Closing this Issue as it was addressed in PR #101
